envoy/network/filter.h

```diff
@@ -330,6 +331,20 @@ class ListenerFilter {
    * @return status used by the filter manager to manage further filter iteration.
    */
   virtual FilterStatus onAccept(ListenerFilterCallbacks& cb) PURE;
+
+  /**
+   * Called when data read from the connection. If the filter chain doesn't get
+   * enough data, the filter chain can be stopped, then waiting for more data.
+   * @param buffer the buffer of data.
+   * @return status used by the filter manager to manage further filter iteration.
+   */
+  virtual FilterStatus onData(Network::ListenerFilterBuffer& buffer) PURE;
+
+  /**
+   * Return the size of data the filter want to inspect from the connection.
+   * @return the size of data inspect from the connection. 0 means filter needn't any data.
```

Comment:
nit: maximum number of bytes of the data consumed by the filter. 0 means filter does not need any data.

Reply:
I noticed this gets updated in the proxy protocol filter. This might be confusing since it looks like a constant but in fact gets changed during `onData`?

Reply:
Yes, this can be changed, like proxy protocol has a variable header. I will clarify it.